Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -972,16 +973,20 @@ void ABenchmarkGymGameModeBase::UpdateAndCheckTotalActorCounts()
 		{
 			// Check for test failure
 			const FExpectedActorCountConfig& ExpectedActorCount = ExpectedActorCounts[ActorClass];
-			bActorCountFailureState = TotalActorCount < ExpectedActorCount.MinCount || TotalActorCount > ExpectedActorCount.MaxCount;
-			if (bActorCountFailureState && !bHasActorCountFailed)
+			bool bActorCountFailed = TotalActorCount < ExpectedActorCount.MinCount || TotalActorCount > ExpectedActorCount.MaxCount;
+			if (bActorCountFailed)
 			{
-				bHasActorCountFailed = true;
-				NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("%s: Unreal actor count check. ObjectClass %s, MinCount %d, MaxCount %d, ActualCount %d"),
-					*NFRFailureString,
-					*ActorClass->GetName(),
-					ExpectedActorCount.MinCount,
-					ExpectedActorCount.MaxCount,
-					TotalActorCount);
+				bActorCountFailureState = true;
+				if (!bHasActorCountFailed)
```

Comment:
Minor: `bLoggedActorCountFailed`
Quite difficult to read with so many similar names.
Alternatively could we just return after logging instead?

Reply:
Will rework to remove `bActorCountFailed`
Not sure what returning early would help with though

Reply:
Maybe I am reading it wrong but it seems we can only log "Unreal actor count check" once. So rather than using a bool to gate that this logs only once, just return instead?
```
...
	for (const auto& ActorCountPair : TempTotalActorCounts)
	{
		...
		if (bIsReadyToConsiderActorCount)
		{
			// Check for test failure
			const FExpectedActorCountConfig& ExpectedActorCount = ExpectedActorCounts[ActorClass];
			if (TotalActorCount < ExpectedActorCount.MinCount || TotalActorCount > ExpectedActorCount.MaxCount)
			{
				bActorCountFailureState = true;
				bHasActorCountFailed = true;
				NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("%s: Unreal actor count check. ObjectClass %s, MinCount %d, MaxCount %d, ActualCount %d"),
						*NFRFailureString,
						*ActorClass->GetName(),
						ExpectedActorCount.MinCount,
						ExpectedActorCount.MaxCount,
						TotalActorCount);
				return;
			}
		}

Reply:
We want `bActorCountFailureState` to be the current state of failre, and the `bHasActorCountFailed` to store if the test has ever failed. So we need to continue to do the check regardless of the current state of `bHasActorCountFailed`

Reply:
And yes - to prevent log spam - we only output the failure message once.

Reply:
Beyond the scope of this ticket but I'd love to see this broken up a bit in a few different functions, its hard work reading this, but maybe its just me.